doc/adrs/XXX_add_support_for_backend-driven_diagram_layout_customization.adoc

```diff
@@ -0,0 +1,185 @@
+- Easier integration with rendering and persistence layers.
+- Better performance and extensibility.
+
+== Decision
+
+We introduce an API mechanism in `DiagramCreationService` to allow post-processing of diagrams after they are created or updated.
+This is done through a new interface, `IDiagramPostProcessor`, that lets us plug in custom logic easily.
+
+The goal is to make it possible to adjust a diagram (layout and/or semantic changes, etc.) without modifying the core computation.
+
+This approach is inspired by a similar mechanism already used for forms (`IFormPostProcessor`), and helps keep things simple and consistent.
+
+
+
+=== API for custom diagram logic Providers
+
+A new backend API is introduced to support domain-specific or diagram-type-specific strategies.
+Furthermore, the specifier can define which specific actions (`IDiagramInput` interface) should trigger these custom logics, such as `EditLabelInput`, `DeleteFromDiagramInput`, etc.
+This brings more flexibility to specifiers to adapt diagram/layout behavior to the needs of their diagrams or domain logic.
+
+==== IDiagramPostProcessor interface
+
+[source,java]
+----
+public interface IDiagramPostProcessor {
+
+    /**
+     * Applies custom logic to a diagram after creation or update.
+     *
+     * @param diagram The diagram to process
+     * @param diagramDescription Optional diagram description (if already available)
+     * @param diagramContext Optional context containing triggering information
+     * @return The modified diagram
+     */
+    Diagram postProcess(Diagram diagram, Optional<DiagramDescription> diagramDescription, Optional<IDiagramContext> diagramContext);
+}
+----
+
+=== DiagramPostProcessorRegistry
+
+[source,java]
+----
+public class DiagramPostProcessorRegistry {
+
+    private final List<IDiagramPostProcessor> postProcessors;
+
+    /**
+     * Applies all registered post-processors to the given diagram.
+     *
+     * @param diagram The diagram to process
+     * @param diagramDescription Optional diagram description (if already available)
+     * @param diagramContext Optional context containing triggering information
+     * @return The modified diagram
+     */
+    public Diagram applyPostProcessors(Diagram diagram, Optional<DiagramDescription> diagramDescription, Optional<IDiagramContext> diagramContext);
+}
+----
+
+==== Integration into DiagramCreationService
+
+[source,java]
+----
+private final DiagramPostProcessorRegistry diagramPostProcessorRegistry;
+
+    /**
+     * Applies custom post-processing logic to the given diagram.
+     *
+     * @param diagram The diagram to process
+     * @param diagramDescription Optional diagram description (if already available)
+     * @param diagramContext Optional context containing triggering information
+     * @return The modified diagram
+     */
+    private Diagram postProcess(Diagram diagram, Optional<DiagramDescription> diagramDescription, Optional<IDiagramContext> diagramContext);
+----
```

Comment:
This could simpler:

public class DiagramCreationService implements IDiagramCreationService {

    private final List<IDiagramPostProcessor> diagramPostProcessors;

    private Diagram doRender(...) {
        ...
        var diagram = ...
        for (var diagramPostProcessor: this.diagramPostProcessors) {
            diagram = diagramPostProcessor.postProcess(...);
        }
        return diagram;
    }
}